File: Java/AA_Lab.java
import java.util.*;
class Node{
    int data;
    Node next;

    Node(int d){
        this.data=d;
        this.next=null;
    }
}

class Stack{

    Node head;

    public void push(int d){
        Node newnode = new Node(d);
        if (head==null){
            head=newnode;
        }
        else{
            newnode.next=head;
            head=newnode;
        }
    }

    public void pop(){
        if(head==null){
            System.out.println("underflow");
            return;
        }
        System.out.println(head.data+" Item Poped");
        head=head.next;
    }

    public void print(){
        if(head==null){
            System.out.println("underflow ");
            return;
        }
        Node temp=head;
        while(temp!=null){
            System.out.println(temp.data);
            temp=temp.next;
        }
    }

    public void peek(){
        if(head==null){
            System.out.println("underflow");
            return;
        }
        System.out.println(head.data+" Peek Item");
    }
}
public class AA_Lab{
    public static void main(String[] args) {
        Stack s1=new Stack();
        Scanner in = new Scanner(System.in);
        int d;
        while(true){
            System.out.println("""
                    1.push
                    2.pop
                    3.peek
                    4.print 
                    5.Exit
                    """);
            d=in.nextInt();
            if(d==5){
                break;
            }
            if(d==1){
                System.out.println("Enter Number: ");
                int data=in.nextInt();
                s1.push(data);
            }
            if(d==2){
                s1.pop();
            }
            if(d==3){
                s1.peek();
            }
            if(d==4){
                s1.print();
            }
        }
    }
}

File: AdvancedAlgorithm/AA_Lab.py
class Node:
    def __init__(self,d):
        self.data=d
        self.next=None
        
class Stack:
    
    def __init__(self):
        self.head=None
        
    def push(self,d):
        Newnode=Node(d)
        if (self.head==None):
            self.head=Newnode
        else:
            Newnode.next=self.head
            self.head=Newnode
    
    def pop(self):
        if(self.head==None):
            print("Underflow")
            return
        self.head=self.head.next
        
    def top(self):
        if(self.head==None):
            print("Underflow")
            return
        print(self.head.data)
          
    def display(self):
        if(self.head==None):
            print("Underflow")
            return
        temp=self.head
        while(temp!=None):
            print(temp.data)
            temp=temp.next
            
s1=Stack()
while (True):
    print("""
          Enter Choice-
          1.Push
          2.Pop
          3.Top
          4.Print
          5.Exit\n
          """)
    d=int(input())

    if(d==1):
        print("Enter Data: ")
        data=int(input())
        s1.push(data)
    elif(d==2):
        s1.pop()
    elif(d==3):
        s1.top()
    elif(d==4):
        s1.display()
    elif(d==5):
        break
    else:
        print("Enter correct choice")